app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,27 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+    @params = params
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to user_post_path(id: @post.id, user_id: @post.author_id), notice: 'Post was successfully created.'
+    else
+      render :new, alert: 'Error occurred, Post not saved'
+    end
+  end
```

Comment:
- [ ] [optional] It would have been nicer using `@current_user` created in the application controller to create your posts comments and likes this will reduce the amount of code you use.
```suggestion
  def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end

```